production/kubernetes/agent-bare.yaml

```diff
@@ -61,22 +76,33 @@ spec:
         - -config.file=/etc/agent/agent.yaml
         command:
         - /bin/agent
-        env:
-        - name: HOSTNAME
-          valueFrom:
-            fieldRef:
-              fieldPath: spec.nodeName
         image: grafana/agent:v0.23.0
         imagePullPolicy: IfNotPresent
-        name: agent
+        name: grafana-agent
         ports:
-        - containerPort: 12345
+        - containerPort: 80
           name: http-metrics
         volumeMounts:
+        - mountPath: /var/lib/agent
+          name: agent-wal
         - mountPath: /etc/agent
           name: grafana-agent
       serviceAccount: grafana-agent
       volumes:
       - configMap:
           name: grafana-agent
         name: grafana-agent
+  updateStrategy:
+    type: RollingUpdate
+  volumeClaimTemplates:
+  - apiVersion: v1
+    kind: PersistentVolumeClaim
+    metadata:
+      name: agent-wal
+      namespace: ${NAMESPACE}
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      resources:
+        requests:
+          storage: 5Gi
```

Comment:
(We might want to tell people to increase this according to how many metrics they expect to send)